SwiftCompilerSources/Sources/Optimizer/Utilities/EscapeUtils.swift

```diff
@@ -420,6 +420,41 @@ fileprivate struct EscapeWalker<V: EscapeVisitor> : ValueDefUseWalker,
           return .continueWalk
         }
         return isEscaping
+
+      case .AtomicLoad:
+        // Treat atomic loads as regular loads and just walk down their uses.
+        guard followLoads(at: path) else {
```

Comment:
nitpick: I find it confusing if `guard` is used in an inverted meaning. Usually it should "protect" the following code from "bad things". But `followLoads` is actually the "bad thing". IMO it's more readable to write:
```
  if !followLoads(at: path) {
    return .continueWalk
  }
```